partner-chains-cli/src/smart_contracts.rs

```diff
@@ -24,3 +25,65 @@ pub fn runtime_config_arguments(
 			payment_signing_key_path
 	)
 }
+
+pub fn check_for_kupo_ogmios_connection_error(
+	response: &str,
+	sidechain_main_cli_resources: &SidechainMainCliResources,
+) -> anyhow::Result<()> {
+	if response.contains("ECONNREFUSED") {
```

Comment:
Unfortunately, this is wrong as well. Please test locally with recent sidechain-main-cli.

Reply:
This is an attempt to fix the error in a wrong place. Instead of us, it should be smart contracts cli to not throw out cryptic errors

Reply:
I've tested this earlier and retested it now and it correctly produces:
```
Error: Failed to connect to Ogmios at http://localhost:1337. Please check connection configuration and try again.
```
As for where this should be fixes - I agree that smart cotracts cli should improve its errors, but in the meantime we've had the requirement to provide a work-around for that. We can clean this up when the CLI improves its messaging

Reply:
I've tested prepare-configuration command with wrong ogmios port and this has not been triggered. sidechain-main-cli error does not contain ECONNREFUSED.

Reply:
@LGLO Sorry, I can't reproduce it on my machine, I get the correct error for both prepare-configuration and setup-main-chain-state. But I do know that sometimes the error message from CLI is different (because the error returned is wrapped somehow), so maybe that's not determinate?

Reply:
We've decided to scrap this check alltogether